lib/API/MP/v1/CustomFields.php

```diff
@@ -64,15 +64,20 @@ public function getSubscriberFields(): array {
   }
 
   public function addSubscriberField(array $data = []): array {
-    $customField = CustomField::createOrUpdate($this->customFieldsDataSanitizer->sanitize($data));
-    $errors = $customField->getErrors();
-    if (!empty($errors)) {
-      throw new APIException('Failed to save a new subscriber field ' . join(', ', $errors), APIException::FAILED_TO_SAVE_SUBSCRIBER_FIELD);
+    try {
+      $customField = $this->customFieldsRepository->createOrUpdate($this->customFieldsDataSanitizer->sanitize($data));
+    } catch (\Exception $e) {
+      throw new APIException('Failed to save a new subscriber field ' . join(', ', $e->getMessage()), APIException::FAILED_TO_SAVE_SUBSCRIBER_FIELD);
     }
-    $customField = CustomField::findOne($customField->id);
-    if (!$customField instanceof CustomField) {
+    $customField = $this->customFieldsRepository->findOneById($customField->getId());
```

Comment:
Why do we need to get the CustomFieldEntity again? I believe it was already returned by $this->customFieldsRepository->createOrUpdate().

Reply:
This is the same as the old code was. So I left it there. We can remove it I guess. We usually did it to get all the fields that are generated by the database like `id` and `created_at`. But I am looking at Doctrine and it does fill the data automatically so I can remove this line.

Reply:
:+1: